index.bs

```diff
@@ -218,11 +219,40 @@ partial interface XRSession {
 };
 </script>
 
-{{XRSession}} is extended to contain the {{XRSession/initiateRoomCapture}} method which, if supported, will ask the [=XRSession/XR device=] to capture the current room layout. It is up to the [=XRSession/XR device=] if this will replace or augment the [=XRSession/set of tracked planes=]. The user agent MAY also ignore this call, for instance if it doesn't support a manual room capture more or if it determines that the room is already set up.
-If the {{XRSession/initiateRoomCapture}} method is called more than once for a given {{XRSession}}, the user agent MUST throw {{InvalidStateError}}.
-If [=plane-detection=] feature descriptor is not [=list/contain|contained=] in the {{XRSession}}'s [=XRSession/XR device=]'s [=XR device/list of enabled features=] for the {{XRSession}}'s [=XRSession/mode=], [=exception/throw=] a {{NotSupportedError}}.
+{{XRSession}} is extended to contain associated <dfn for=XRSession>set of tracked planes</dfn>, which is initially empty. The elements of the set will be of {{XRPlane}} type.
 
-{{XRSession}} is also extended to contain associated <dfn for=XRSession>set of tracked planes</dfn>, which is initially empty. The elements of the set will be of {{XRPlane}} type.
+{{XRSession}} is extended to contain a boolean <dfn for=XRSession>room capture completed</dfn>, which is initially false.
+
+If [=XRSession/XR device=] supports manual capture, it has an async <dfn for="XR device">room capture method</dfn> which returns a boolean.
+
+{{XRSession}} is also extended to contain the {{XRSession/initiateRoomCapture}} method which, if supported, will ask the [=XRSession/XR device=] to capture the current room layout. It is up to the [=XRSession/XR device=] if this will replace or augment the [=XRSession/set of tracked planes=].
+
+<div class="algorithm" data-algorithm="initiate-room-capture">
+When this method is invoked, the user agent MUST run the following steps:
+  1. Let |session| be [=this=]
+  1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of |session|.
+  1. If the {{XRSession/ended}} value of |session| is `true`, [=reject=] |promise| with a "{{InvalidStateError}}" {{DOMException}} and return |promise|.
+  1. If [=XRSession/room capture completed=] is `true`, [=reject=] |promise| with a "{{InvalidStateError}}" {{DOMException}} and return |promise|.
+  1. If [=plane-detection=] feature descriptor is not [=list/contain|contained=] in |session|'s [=XRSession/XR device=]'s [=XR device/list of enabled features=] for |session|'s [=XRSession/mode=], [=reject=] |promise| with a "{{NotSupportedError}}" {{DOMException}} and return |promise|.
+  1. If the |session|'s [=XRSession/XR device=] doesn't support manual capture or if the room is already set up, run the following steps:
```

Comment:
we allow you to re-capture the room

Reply:
`can't run` feels like an error. Maybe `isn't needed`. Though again, that makes me think that I should modify the description of `room capture method` to caveat that it can return `true` if it determines that room capture isn't needed.

Reply:
Done.